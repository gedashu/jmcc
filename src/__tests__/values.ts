import JMCC from '../core';

const jmcc = new JMCC();

describe('значения', () => {
  describe('текст', () => {
    test('с двойными кавычками', () => {
      const source = `
        var a = "string"
      `;

      expect(jmcc.compileToJson(source)).toMatchSnapshot();
    });

    test('с одинарными кавычками', () => {
      const source = `
        var a = 'string'
      `;

      expect(jmcc.compileToJson(source)).toMatchSnapshot();
    });
  });

  describe('число', () => {
    test('целое', () => {
      const source = `
        var a = 1
      `;

      expect(jmcc.compileToJson(source)).toMatchSnapshot();
    });

    test('дробное', () => {
      const source = `
        var a = 0.1
      `;

      expect(jmcc.compileToJson(source)).toMatchSnapshot();
    });

    test('отрицательное', () => {
      const source = `
        var a = -1
      `;

      expect(jmcc.compileToJson(source)).toMatchSnapshot();
    });
  });

  describe('массив', () => {
    test('пустой', () => {
      const source = `
        var a = []
      `;

      expect(jmcc.compileToJson(source)).toMatchSnapshot();
    });

    test('с элементами', () => {
      const source = `
        var a = [1, 2, 3, 4, 5]
      `;

      expect(jmcc.compileToJson(source)).toMatchSnapshot();
    });

    test('с больше 21 элементов', () => {
      const source = `
        var a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22]
      `;

      expect(jmcc.compileToJson(source)).toMatchSnapshot();
    });

    test('вложенный', () => {
      const source = `
        var a = [[1]]
      `;

      expect(jmcc.compileToJson(source)).toMatchSnapshot();
    });
  });

  describe('переменные', () => {
    test('определённые', () => {
      const source = `
        var a = 0
        var b = a
      `;

      expect(jmcc.compileToJson(source)).toMatchSnapshot();
    });

    test('игровые', () => {
      const source = `
        var a = value::location
      `;

      expect(jmcc.compileToJson(source)).toMatchSnapshot();
    });
  });
});
