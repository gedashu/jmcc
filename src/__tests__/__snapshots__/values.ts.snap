// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`значения массив вложенный 1`] = `"Нельзя создавать вложенные массивы"`;

exports[`значения массив пустой 1`] = `
Object {
  "handlers": Array [
    Object {
      "event": "world_start",
      "operations": Array [
        Object {
          "action": "set_variable_create_list",
          "values": Array [
            Object {
              "name": "variable",
              "value": Object {
                "scope": "local",
                "type": "variable",
                "variable": "a",
              },
            },
            Object {
              "name": "values",
              "value": Object {
                "type": "array",
                "values": Array [
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                ],
              },
            },
          ],
        },
      ],
      "position": 0,
      "type": "event",
    },
  ],
}
`;

exports[`значения массив с больше 21 элементов 1`] = `
Object {
  "handlers": Array [
    Object {
      "event": "world_start",
      "operations": Array [
        Object {
          "action": "set_variable_create_list",
          "values": Array [
            Object {
              "name": "variable",
              "value": Object {
                "scope": "local",
                "type": "variable",
                "variable": "a",
              },
            },
            Object {
              "name": "values",
              "value": Object {
                "type": "array",
                "values": Array [
                  Object {
                    "number": 1,
                    "type": "number",
                  },
                  Object {
                    "number": 2,
                    "type": "number",
                  },
                  Object {
                    "number": 3,
                    "type": "number",
                  },
                  Object {
                    "number": 4,
                    "type": "number",
                  },
                  Object {
                    "number": 5,
                    "type": "number",
                  },
                  Object {
                    "number": 6,
                    "type": "number",
                  },
                  Object {
                    "number": 7,
                    "type": "number",
                  },
                  Object {
                    "number": 8,
                    "type": "number",
                  },
                  Object {
                    "number": 9,
                    "type": "number",
                  },
                  Object {
                    "number": 10,
                    "type": "number",
                  },
                  Object {
                    "number": 11,
                    "type": "number",
                  },
                  Object {
                    "number": 12,
                    "type": "number",
                  },
                  Object {
                    "number": 13,
                    "type": "number",
                  },
                  Object {
                    "number": 14,
                    "type": "number",
                  },
                  Object {
                    "number": 15,
                    "type": "number",
                  },
                  Object {
                    "number": 16,
                    "type": "number",
                  },
                  Object {
                    "number": 17,
                    "type": "number",
                  },
                  Object {
                    "number": 18,
                    "type": "number",
                  },
                  Object {
                    "number": 19,
                    "type": "number",
                  },
                  Object {
                    "number": 20,
                    "type": "number",
                  },
                  Object {
                    "number": 21,
                    "type": "number",
                  },
                ],
              },
            },
          ],
        },
        Object {
          "action": "set_variable_append_value",
          "values": Array [
            Object {
              "name": "variable",
              "value": Object {
                "scope": "local",
                "type": "variable",
                "variable": "a",
              },
            },
            Object {
              "name": "values",
              "value": Object {
                "type": "array",
                "values": Array [
                  Object {
                    "number": 22,
                    "type": "number",
                  },
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                ],
              },
            },
          ],
        },
      ],
      "position": 0,
      "type": "event",
    },
  ],
}
`;

exports[`значения массив с элементами 1`] = `
Object {
  "handlers": Array [
    Object {
      "event": "world_start",
      "operations": Array [
        Object {
          "action": "set_variable_create_list",
          "values": Array [
            Object {
              "name": "variable",
              "value": Object {
                "scope": "local",
                "type": "variable",
                "variable": "a",
              },
            },
            Object {
              "name": "values",
              "value": Object {
                "type": "array",
                "values": Array [
                  Object {
                    "number": 1,
                    "type": "number",
                  },
                  Object {
                    "number": 2,
                    "type": "number",
                  },
                  Object {
                    "number": 3,
                    "type": "number",
                  },
                  Object {
                    "number": 4,
                    "type": "number",
                  },
                  Object {
                    "number": 5,
                    "type": "number",
                  },
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                ],
              },
            },
          ],
        },
      ],
      "position": 0,
      "type": "event",
    },
  ],
}
`;

exports[`значения переменные игровые 1`] = `
Object {
  "handlers": Array [
    Object {
      "event": "world_start",
      "operations": Array [
        Object {
          "action": "set_variable_value",
          "values": Array [
            Object {
              "name": "variable",
              "value": Object {
                "scope": "local",
                "type": "variable",
                "variable": "a",
              },
            },
            Object {
              "name": "value",
              "value": Object {
                "game_value": "location",
                "selection": "{\\"type\\":\\"default\\"}",
                "type": "game_value",
              },
            },
          ],
        },
      ],
      "position": 0,
      "type": "event",
    },
  ],
}
`;

exports[`значения переменные определённые 1`] = `
Object {
  "handlers": Array [
    Object {
      "event": "world_start",
      "operations": Array [
        Object {
          "action": "set_variable_value",
          "values": Array [
            Object {
              "name": "variable",
              "value": Object {
                "scope": "local",
                "type": "variable",
                "variable": "a",
              },
            },
            Object {
              "name": "value",
              "value": Object {
                "number": 0,
                "type": "number",
              },
            },
          ],
        },
        Object {
          "action": "set_variable_value",
          "values": Array [
            Object {
              "name": "variable",
              "value": Object {
                "scope": "local",
                "type": "variable",
                "variable": "b",
              },
            },
            Object {
              "name": "value",
              "value": Object {
                "scope": "local",
                "type": "variable",
                "variable": "a",
              },
            },
          ],
        },
      ],
      "position": 0,
      "type": "event",
    },
  ],
}
`;

exports[`значения текст с двойными кавычками 1`] = `
Object {
  "handlers": Array [
    Object {
      "event": "world_start",
      "operations": Array [
        Object {
          "action": "set_variable_value",
          "values": Array [
            Object {
              "name": "variable",
              "value": Object {
                "scope": "local",
                "type": "variable",
                "variable": "a",
              },
            },
            Object {
              "name": "value",
              "value": Object {
                "text": "string",
                "type": "text",
              },
            },
          ],
        },
      ],
      "position": 0,
      "type": "event",
    },
  ],
}
`;

exports[`значения текст с одинарными кавычками 1`] = `
Object {
  "handlers": Array [
    Object {
      "event": "world_start",
      "operations": Array [
        Object {
          "action": "set_variable_value",
          "values": Array [
            Object {
              "name": "variable",
              "value": Object {
                "scope": "local",
                "type": "variable",
                "variable": "a",
              },
            },
            Object {
              "name": "value",
              "value": Object {
                "text": "string",
                "type": "text",
              },
            },
          ],
        },
      ],
      "position": 0,
      "type": "event",
    },
  ],
}
`;

exports[`значения число дробное 1`] = `
Object {
  "handlers": Array [
    Object {
      "event": "world_start",
      "operations": Array [
        Object {
          "action": "set_variable_value",
          "values": Array [
            Object {
              "name": "variable",
              "value": Object {
                "scope": "local",
                "type": "variable",
                "variable": "a",
              },
            },
            Object {
              "name": "value",
              "value": Object {
                "number": 0.1,
                "type": "number",
              },
            },
          ],
        },
      ],
      "position": 0,
      "type": "event",
    },
  ],
}
`;

exports[`значения число отрицательное 1`] = `
Object {
  "handlers": Array [
    Object {
      "event": "world_start",
      "operations": Array [
        Object {
          "action": "set_variable_value",
          "values": Array [
            Object {
              "name": "variable",
              "value": Object {
                "scope": "local",
                "type": "variable",
                "variable": "a",
              },
            },
            Object {
              "name": "value",
              "value": Object {
                "number": -1,
                "type": "number",
              },
            },
          ],
        },
      ],
      "position": 0,
      "type": "event",
    },
  ],
}
`;

exports[`значения число целое 1`] = `
Object {
  "handlers": Array [
    Object {
      "event": "world_start",
      "operations": Array [
        Object {
          "action": "set_variable_value",
          "values": Array [
            Object {
              "name": "variable",
              "value": Object {
                "scope": "local",
                "type": "variable",
                "variable": "a",
              },
            },
            Object {
              "name": "value",
              "value": Object {
                "number": 1,
                "type": "number",
              },
            },
          ],
        },
      ],
      "position": 0,
      "type": "event",
    },
  ],
}
`;
